src/unix/fs.c

```diff
@@ -188,9 +193,38 @@ static ssize_t uv__fs_futime(uv_fs_t* req) {
   }
 
   return r;
+#elif defined(__APPLE__)
+  struct attrlist attr_list;
+  struct timespec times[3];
+  unsigned i = 0;
+
+  bzero(&attr_list, sizeof(attr_list));
+
+  attr_list.bitmapcount = ATTR_BIT_MAP_COUNT;
+
+  if (!isnan(req->btime)) {
+    attr_list.commonattr |= ATTR_CMN_CRTIME;
+
+    times[i++].tv_sec = req->btime;
+    times[i].tv_nsec = (unsigned long)(req->btime * 1000000) % 1000000 * 1000;
```

Comment:
Just curious, is there a reason you're reducing to microseconds here when it can do nanoseconds?